File: 2259_remove_digit_from_number_to_maximize_result/2259_remove_digit_from_number_to_maximize_result.py
def main():
    test0 = ["123", "3"]
    test1 = ["1231", "1"]
    test2 = ["551", "5"]
    tests = [test0, test1, test2]

    for test in tests:
        Solution().removeDigit(test[0], test[1])


class Solution:
    def __init__(self):
        self.winner = None

    def removeDigit(self, number: str, digit: str) -> str:
        number_options = []
        for i, char in enumerate(number):
            if char == digit:
                number_options.append(int(number[0:i] + number[i + 1 :]))
        self.winner = self.largest_number(number_options)

        return self.winner

    def largest_number(self, list_of_nums: list) -> int:
        winner = 0
        for number in list_of_nums:
            if number > winner:
                winner = number

        return str(winner)


if __name__ == "__main__":
    main()

File: 2259_remove_digit_from_number_to_maximize_result/c-solution/2259-remove-digit-from-number-to-maximize-result.c
#include <string.h>
#include <stdio.h>
int main(void)
{
    char *number1 = "123";
    char *number2 = "1231";
    char *number3 = "551";

    char *test1[2] = [ number1, '3' ];
    char *text2[2] = [ number2, '1' ];
    char *text3[2] = [ number3, '1' ];

    char *tests[3] = [ test1, test2, test3 ];
    for (int i = 0; i < 3; i++)
    {
        removeDigit(tests[i][0], tests[i][1]);
    }
}
char *removeDigit(char *number, char digit)
{
    int count = 0;
    while (number[count] != "/0")
    {
        if (number[count] == digit && number[count + 1] > number[count])
        {
        }
    }
}